plugins/hydra_rq_launcher/hydra_plugins/hydra_rq_launcher/_core.py

```diff
@@ -0,0 +1,146 @@
+        os.chdir(work_dir)
+
+    setup_globals()
+    Singleton.set_state(singleton_state)
+
+    sweep_config = config_loader.load_sweep_config(config, list(overrides))
+    with open_dict(sweep_config):
+        sweep_config.hydra.job.id = redis_job_id
+        sweep_config.hydra.job.num = idx
+    HydraConfig.instance().set_config(sweep_config)
+
+    ret = run_job(
+        config=sweep_config,
+        task_function=task_function,
+        job_dir_key="hydra.sweep.dir",
+        job_subdir_key="hydra.sweep.subdir",
+    )
+
+    return ret
+
+
+def launch(
+    launcher: RQLauncher, job_overrides: Sequence[Sequence[str]], initial_job_idx: int,
+) -> JobReturn:
+    """
+    :param job_overrides: a List of List<String>, where each inner list is the arguments for one job run.
+    :param initial_job_idx: Initial job idx in batch.
+    :return: an array of return values from run_job with indexes corresponding to the input list indexes.
+    """
+    setup_globals()
+    assert launcher.config is not None
+    assert launcher.config_loader is not None
+    assert launcher.task_function is not None
+
+    configure_log(launcher.config.hydra.hydra_logging, launcher.config.hydra.verbose)
+    sweep_dir = Path(str(launcher.config.hydra.sweep.dir))
+    sweep_dir.mkdir(parents=True, exist_ok=True)
+
+    # RQ configuration
+    rq_cfg = launcher.rq
+    connection = Redis(**rq_cfg["redis"])
+    queue = Queue(name=rq_cfg["queue"], connection=connection, serializer=cloudpickle)
+    work_dir = hydra.utils.get_original_cwd() if rq_cfg["change_work_dir"] else None
+
+    jobs = []
+    singleton_state = Singleton.get_state()
+
+    log.info(
+        "RQ Launcher ({}) is launching {} jobs".format(
+            ",".join([f"{k}={v}" for k, v in rq_cfg.items()]), len(job_overrides),
+        )
+    )
+    log.info("Enqueueing jobs, sweep output dir : {}".format(sweep_dir))
+    for idx, overrides in enumerate(job_overrides):
+        description = " ".join(filter_overrides(overrides))
+        log.info(f"\t#{idx} : {description}")
+
+        enqueue_keywords = OmegaConf.to_container(rq_cfg["enqueue"], resolve=True)
+        if enqueue_keywords["job_id"] is None:
+            enqueue_keywords["job_id"] = str(uuid.uuid4())
+        if enqueue_keywords["description"] is None:
+            enqueue_keywords["description"] = description
+
+        job = queue.enqueue(
+            execute_job,
+            idx=initial_job_idx + idx,
+            overrides=overrides,
+            config_loader=launcher.config_loader,
+            config=launcher.config,
+            task_function=launcher.task_function,
+            singleton_state=singleton_state,
+            redis_job_id=enqueue_keywords["job_id"],
+            work_dir=work_dir,
+            **enqueue_keywords,
+        )
+        jobs.append(job)
+
+        log.info(f"Enqueued {job.get_id()}")
+
+    log.info("Finished enqueuing")
+    if rq_cfg["quit_after_enqueue"]:
+        quit()
```

Comment:
I am tempted to not support this.
the plugin should definitely not call quit in the middle of processing.
if anything it should raise a special exception that indicates to Hydra that this plugin launched asynchronously and that it should finish the run.

Reply:
Agreed that quitting directly isn't clean, so I went for implementing a custom exception. I've found it very helpful to have the option of to not wait for results, e.g., in my use cases so far results are stored to disk on remote servers and I don't need the main processes waiting